lib/ansible/modules/apt_repository.py

```diff
@@ -72,8 +72,12 @@
     filename:
         description:
             - Sets the name of the source list file in sources.list.d.
-              Defaults to a file name based on the repository source url.
-              The .list extension will be automatically added.
+            - Defaults to a file name based on the repository source URL (for example, https://user:pass@packagecloud.io/linz/prod/ubuntu/)
```

Comment:
Could we use a link to a non-existing repository instead? Maybe `https://username:password@packages.example.com/org/repo/distro/`? I don't know if packagecloud has any I/O limits on access to a repository, but it would probably be a good idea to avoid undue traffic there.

Reply:
instead of a lot of details most users don't care about, we should ensure to return the file name produced (and update return docs)

Reply:
```diff --git a/lib/ansible/modules/apt_repository.py b/lib/ansible/modules/apt_repository.py
index f9a0cd9195..59270109ce 100644
--- a/lib/ansible/modules/apt_repository.py
+++ b/lib/ansible/modules/apt_repository.py
@@ -688,15 +688,16 @@ def main():
     sources_after = sourceslist.dump()
     changed = sources_before != sources_after

-    if changed and module._diff:
-        diff = []
-        for filename in set(sources_before.keys()).union(sources_after.keys()):
-            diff.append({'before': sources_before.get(filename, ''),
-                         'after': sources_after.get(filename, ''),
-                         'before_header': (filename, '/dev/null')[filename not in sources_before],
-                         'after_header': (filename, '/dev/null')[filename not in sources_after]})
-    else:
-        diff = {}
+    diff = []
+    if changed:
+        sources_added = set(sources_after.keys()).difference(sources_before.keys())
+        sources_removed = set(sources_before.keys()).difference(sources_after.keys())
+        if module._diff:
+            for filename in set(sources_added, sources_removed):
+                diff.append({'before': sources_before.get(filename, ''),
+                             'after': sources_after.get(filename, ''),
+                             'before_header': (filename, '/dev/null')[filename not in sources_before],
+                             'after_header': (filename, '/dev/null')[filename not in sources_after]})

     if changed and not module.check_mode:
         try:
@@ -728,7 +729,7 @@ def main():
             revert_sources_list(sources_before, sources_after, sourceslist_before)
             module.fail_json(msg=to_native(ex))

-    module.exit_json(changed=changed, repo=repo, state=state, diff=diff)
+    module.exit_json(changed=changed, repo=repo, sources_added=sources_added, sources_remove=sources_removed, state=state, diff=diff)


 if __name__ == '__main__':```
and updating `RETURN`  to document these and the existing fields would probably go much farther than trying to explain the naming logic

Reply:
Got it @bcoca. Thanks for the feedback. I will document and implement this.